src/btree/bt_sync.c

```diff
@@ -148,16 +149,27 @@ __sync_delete_obsolete_ref(WT_SESSION_IMPL *session, WT_REF *ref)
         return (0);
     }
```

Comment:
Also there is no change to the code to bypass checkpoint cleanup if we are under cache pressure. I guess that is acceptable for now.